### PR TITLE
Right Stick controls IR pointer

### LINKED ISSUE #79: No longer able to map IR to right analog stick - WiimoteNew.ini is ignored
In previous builds I was able to use a workaround to map my right analog stick, to the wii cursor, instead of an actual mouse. This would work for all 4 controllers for all players, and allowed a very good control of most of the wii games. In order to make this work properly, I have edited the file:
Retroarch\saves\User\Config\WiimoteNew.ini to contain the following:

```
IR/Up = `Libretro/0/Analog:Y1-`
IR/Down = `Libretro/0/Analog:Y1+`
IR/Left = `Libretro/0/Analog:X1-`
IR/Right = `Libretro/0/Analog:X1+`
IR/Width = 11.000000000000000
IR/Height = 11.000000000000000
IR/Dead Zone = 4.000000000000000
IR/Relative Input = True
IR/Auto-Hide = True

`````

To avoid the core to overrwrite my custom configfile, I made the file read-only so that my settings would stay put. This would werk perfectly, as the core would create a WiimoteNew.ini.xxx file instead, but just ignored it and used my own config instead. With the latest core update this, does no longer work and I am unable to use my right stick, as the IR pointer in wii games. As you can imagine this is a very usefull configuration that allows a lot of Wii games to be played with your controllers instead of a mouse. 

So please allow the overrided WiimoteNew.ini file to be read again, or make it possible to use the controller menu, to map the right analog stick to the IR controllers. Hope someone is able to help me out here.

### PR DESCRIPTION
**Pull request by goedzo**
This allows the Wii/IR pointer to be controlled by the right analog stick.  Fixes #79

### CODE REVIEW

**Comment by flyinghead** [comment]
With this change, the IR pointer cannot be controlled by a mouse/pointer anymore. This is not going to work for light gun games and such.
A core option to select between mouse or right analog stick would be better. One of the Nintendo DS cores has such an option.

**Comment by goedzo** [comment]
Interesting suggestion however i am new to this core for development. Could you give me some pointers on where in the code i can add an extra core option? I will then try to adapt the code to use it.

**Comment by flyinghead** [comment]
You need to define a new core option in ```Source/Core/DolphinLibretro/Options.h/cpp```.
Then use this option's value in ```Source/Core/DolphinLibretro/Input.cpp```to bind the IR pointer to the mouse or right analog stick.

**Comment by goedzo** [comment]
Thanks! Implemented this now as a core option. Please merge now :).

File: Source/Core/DolphinLibretro/Options.cpp [hunk 1]
```diff
@@ -148,7 +148,8 @@ bool Option<T>::Updated()
 Option<int> efbScale("dolphin_efb_scale", "EFB Scale", 1,
                      {"x1 (640 x 528)", "x2 (1280 x 1056)", "x3 (1920 x 1584)", "x4 (2560 * 2112)",
                       "x5 (3200 x 2640)", "x6 (3840 x 3168)"});
-Option<LogTypes::LOG_LEVELS> logLevel("dolphin_log_level", "Log Level",
+Option<int> irMode("dolphin_ir_mode", "IR Mode", 1,
+                     {"Right Stick controls pointer", "Mouse controls pointer"});Option<LogTypes::LOG_LEVELS> logLevel("dolphin_log_level", "Log Level",
                                       {{"Info", LogTypes::LINFO},
```
Comment (flyinghead) [inline Source/Core/DolphinLibretro/Options.cpp:154]: Missing `\n`

File: Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp [hunk 2]
```diff
@@ -30,13 +32,25 @@ Cursor::Cursor(const std::string& name_) : ControlGroup(name_, GroupType::Cursor
...
+    boolean_settings.emplace_back(std::make_unique<BooleanSetting>(_trans("Relative Input"), true));
+    boolean_settings.emplace_back(std::make_unique<BooleanSetting>(_trans("Auto-Hide"), true));
+  }
+  else {
+    // Mouse controls IR
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Center"), 0.5));
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Width"), 0.5));
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Height"), 0.5));
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Dead Zone"), 0, 0, 20));
+    boolean_settings.emplace_back(std::make_unique<BooleanSetting>(_trans("Relative Input"), true));
```
Comment (flyinghead) [inline Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp]: Relative input was false for screen pointer/mouse.
Auto-hide was false.
Any reason to change these values?

**Review by flyinghead — CHANGES_REQUESTED** [review]

File: Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp (hunk 2 above)
Comment (goedzo) [reply Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp]: Nope, just  copy and paste error here.

File: Source/Core/DolphinLibretro/Options.cpp (hunk 1 above)
Comment (goedzo) [reply Source/Core/DolphinLibretro/Options.cpp:154]: Fixed

**Comment by goedzo** [comment]
Fixed all comments. Thanks for the review!

**Review by flyinghead — APPROVED** [review]
Looks good to me

**Pull request merged**